modules/core/eks/stack.py

```diff
@@ -121,16 +135,21 @@ def __init__(
         cluster_admin_role.add_to_principal_policy(iam.PolicyStatement.from_json(cluster_admin_policy_statement_json_2))
         cluster_admin_role.add_to_principal_policy(iam.PolicyStatement.from_json(cluster_admin_policy_statement_json_3))
 
+        # KMS key for Envelope encryption
+        secrets_key = kms.Key(self, "SecretsKey")
+
         # Creates an EKS Cluster
         eks_cluster = eks.Cluster(
             self,
             "cluster",
             vpc=self.vpc,
+            vpc_subnets=[ec2.SubnetSelection(subnets=self.private_subnets)],
             cluster_name=f"addf-{self.deployment_name}-{self.module_name}-cluster",
             masters_role=cluster_admin_role,
-            endpoint_access=eks.EndpointAccess.PUBLIC,
+            endpoint_access=eks.EndpointAccess.PRIVATE if self.isolated_subnet_ids else eks.EndpointAccess.PUBLIC,
             version=eks.KubernetesVersion.of(str(self.eks_compute_config["eks_version"])),
             default_capacity=0,
+            secrets_encryption_key=secrets_key,
```

Comment:
If the KMS key was null, then this is backward compatible?  But the way the key is created causes a redeploy of the cluster?